build.sbt

```diff
@@ -1,17 +1,57 @@
-name := "grpc-sample"
+import sbt.Keys.libraryDependencies
+import grpc.sbt.sample.Versions
 
-version := "1.0"
+val macwire = "com.softwaremill.macwire" %% "macros" % "2.2.5" % "provided"
+val scalaTest = "org.scalatest" %% "scalatest" % "3.0.1" % Test
 
-scalaVersion := "2.11.11"
 
+lazy val `helloscala` = (project in file("."))
+  .settings(
+    organization in ThisBuild := "com.example",
+    version in ThisBuild := "1.0-SNAPSHOT",
+    // the Scala version that will be used for cross-compiled libraries
+    scalaVersion in ThisBuild := "2.11.8"
+  )
+  .aggregate(`helloscala-grpc-server`, `helloscala-lagom-client`)
 
-PB.targets in Compile := Seq(
-  PB.gens.java -> (sourceManaged in Compile).value,
-  scalapb.gen(javaConversions = true) -> (sourceManaged in Compile).value
-)
+lazy val `helloscala-grpc-server` = (project in file("helloscala-grpc-server"))
+  .enablePlugins(ProtocPlugin)
+  .settings(grpcCommon)
+  .settings(scalapbSettings("helloscala-grpc-server", forServer = true))
+
+lazy val `helloscala-lagom-client` = (project in file("helloscala-lagom-client"))
+  .enablePlugins(ProtocPlugin)
+  .settings(grpcCommon)
+  .settings(scalapbSettings("helloscala-lagom-client"))
+  .settings(
+    libraryDependencies ++= Seq(
+      lagomScaladslClient,
+      lagomScaladslApi,
+      lagomLogback
+    )
+  )
+
+
+// ------------------------------------------------------------------------------------------
+
+// TODO: generate grpc for java in any platform. See https://gitter.im/trueaccord/ScalaPB?at=59230464fa63ba2f766aba1b
+def scalapbSettings(projectFolder: String, forServer: Boolean = false) = {
+  Seq(
+    PB.targets in Compile := Seq(
+      PB.gens.java -> (sourceManaged in Compile).value,
+      scalapb.gen(
+        javaConversions = true,
+        grpc = forServer
+      ) -> (sourceManaged in Compile).value
```

Comment:
With the boolean `grpc` in scalapb it is possible to enable/disable the creation of the grpc-related code.
In our case we want the pure-grpc project to get that generated by the plugin and the lagom-grpc client project  to obtain it from lagom so the `grpc` flag for the scalapb plugin is only enabled in the server.
<img width="329" alt="screen shot 2017-05-25 at 18 40 33" src="https://cloud.githubusercontent.com/assets/762126/26460414/d41675c4-4179-11e7-9d59-dedb54c91d68.png">
<img width="349" alt="screen shot 2017-05-25 at 18 40 38" src="https://cloud.githubusercontent.com/assets/762126/26460415/d41b132c-4179-11e7-86ae-d6dc2e2676f4.png">

See in the screenshots how each project obtains a different output.

NOTE: both have the java code for the protobuf classes because of `PB.gens.java `. I haven't tested anything related to java code.